runelite-client/src/main/java/net/runelite/client/plugins/microbot/lunartanner/TanLeatherScript.java

```diff
@@ -0,0 +1,102 @@
+    private enum State {
+        TANNING,
+        BANKING
+    }
+
+    private State currentState = State.TANNING;
+
+    public boolean run(TanLeatherConfig config) {
+        startTime = System.currentTimeMillis();
+        int unprocessedItemPrice = Microbot.getItemManager().search(config.ITEM().getName()).get(0).getPrice();
+        int processedItemPrice = Microbot.getItemManager().search(config.ITEM().getFinished()).get(0).getPrice();
+        profitPerHide = processedItemPrice - unprocessedItemPrice;
+
+        mainScheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            if (!super.run()) return;
+            if (!Microbot.isLoggedIn()) return;
+            try {
+                if (Microbot.pauseAllScripts) return;
+                if (Rs2Inventory.hasItem(config.ITEM().getName(), true)) {
+                    Rs2Magic.cast(MagicAction.TAN_LEATHER);
+                    sleepUntilOnClientThread(() -> Rs2Inventory.hasItem(config.ITEM().getFinished()));
+                    hidesTanned++;
+                } else {
+                    bank(config);
+                }
+            } catch (Exception ex) {
+                System.out.println(ex.getMessage());
+            }
+        }, 0, 50, TimeUnit.MILLISECONDS);
+        return true;
+    }
+
+    // Calculate the profit and display it
+    private void calculateProfitAndDisplay(TanLeatherConfig config) {
+        double elapsedHours = (System.currentTimeMillis() - startTime) / 3600000.0;
+        int hidesPerHour = (int) (hidesTanned / elapsedHours);
+        int totalProfit = profitPerHide * (int)hidesTanned;
+        int profitPerHour = profitPerHide * hidesPerHour;
+
+        // Format the message
+        combinedMessage = config.ITEM().getFinished() + ": " +
+                QuantityFormatter.quantityToRSDecimalStack((int) hidesTanned) + " (" +
+                QuantityFormatter.quantityToRSDecimalStack(hidesPerHour) + "/hr) | " +
+                "Profit: " + QuantityFormatter.quantityToRSDecimalStack(totalProfit) + " (" +
+                QuantityFormatter.quantityToRSDecimalStack(profitPerHour) + "/hr)";
+    }
+
+    // Bank the finished hide
+    private void bank(TanLeatherConfig config) {
+        if (currentState != State.BANKING) {
+            currentState = State.BANKING;
+            if (!Rs2Bank.isOpen()) {
+                Rs2Bank.openBank();
+                sleepUntilOnClientThread(Rs2Bank::isOpen);
+            }
+
+            Rs2Bank.depositAll(config.ITEM().getFinished());
+            sleepUntilOnClientThread(() -> !Rs2Inventory.hasItem(config.ITEM().getFinished()));
+
+            if (Rs2Bank.hasItem(config.ITEM().getName())) {
+                Rs2Bank.withdrawAll(config.ITEM().getName());
+                sleepUntilOnClientThread(() -> Rs2Inventory.hasItem(config.ITEM().getName()));
+            } else {
+                shutdown();
+                return;
+            }
+
+            Rs2Bank.closeBank();
```

Comment:
There is already a  sleepUntilOnClientThread(() -> !Rs2Bank.isOpen()); inside closeBank()